### ISSUE TITLE
Model training stalls forever after just a few batches. 

### ISSUE THREAD

**Issue by not-Ian** [tensorflow/tensorflow#45716]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution: Windows 10.0.19041 Build 19041
- TensorFlow installed via pip
- TensorFlow version: 2.4.0-rc4
- TensorFlow Git version: v2.4.0-rc3-20-g97c3fef64ba 
- Python version: 3.8.5
- CUDA/cuDNN version: CUDA 11.0, cuDNN 8.0.4
- GPU model and memory: Nvidia RTX 3090, 24GB RAM

**Describe the current behavior**
Model training regularly freezes for large models. 
Sometimes the first batch or so works, but then just a few batches later and training seems stuck in a loop. From my activity monitor, I see GPU CUDA use hovering around 100%. This goes on for minutes or more, with no more batches being trained.

I don't see an OOM error, nor does it seem like I'm hitting memory limits in activity monitor or `nvidia-smi`.

**Describe the expected behavior**
I would expect the first batch to take a bit longer, then any subsequent batches to take less than <1s. Never have a random batch take minutes (after previous batches took <1s) or stall forever.

**Standalone code to reproduce the issue**
Run through all the cells in the notebook shared below to initialize the model, then run the final cell just a few times. Eventually it will hang and never finish.

https://github.com/not-Ian/tensorflow-bug-example/blob/main/tensorflow%20error%20example.ipynb

**Other info / logs**
My intuition:
Smaller models train quickly as expected, however I think even then they eventually stall out after training many, many batches.
I had another similar, small VAE like in my example that trained for 5k-10k batches overnight before stalling. Could there be a memory leak somewhere?

**Comment by amahendrakar** [comment]
@not-Ian,
Could you please update TensorFlow to the stable version v2.4 and check if you are facing the same issue.

Also, try setting a hard limit on the total GPU memory as per [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and check if it helps. Thanks!

**Comment by not-Ian** [comment]
Thanks for the reply, but no luck. I ran:

`> pip uninstall tensorflow-gpu`

then 

`> pip install tensorflow-gpu==2.4.0`

So now I'm on:

TensorFlow version: 2.4.0
TensorFlow Git version: v2.4.0-rc4-71-g582c8d236cb

Then I included this code at the top of my notebook:

```
gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
  try:
    # setting memory limit to 1024MB * 23 = 23GB. I've got 24, but 23 to be safe.
    tf.config.experimental.set_virtual_device_configuration(
        gpus[0],
        [tf.config.experimental.VirtualDeviceConfiguration(memory_limit=1024 * 23)])
    logical_gpus = tf.config.experimental.list_logical_devices('GPU')
    print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
  except RuntimeError as e:
    # Virtual devices must be set before GPUs have been initialized
    print(e)
```

And training stalled again (100% GPU usage, nothing training) after 2 batches.

**Comment by not-Ian** [comment]
Also, for completion's sake, I got this error:

`2020-12-16 15:50:23.428273: I tensorflow/core/platform/windows/subprocess.cc:308] SubProcess ended with return code: 4294967295`

I read once somewhere about a `ptxas.exe` compiler error when this happens. Something about using the CUDA 11.1 `ptxas.exe` file, as it supports `-arch=sm_86` argument in this call:

`2020-12-16 16:03:52.614874: I tensorflow/stream_executor/gpu/asm_compiler.cc:215] C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.0/bin/ptxas.exe C:\Users\Ian\AppData\Local\Temp\/tempfile-DESKTOP-IDQ189A-1554-3668-5b69c0cfcfd7f -o C:\Users\Ian\AppData\Local\Temp\/tempfile-DESKTOP-IDQ189A-1554-3668-5b69c0cfd038e -arch=sm_86 -v`

Anyways I uninstalled, and reinstalled Tensorflow 2.4.0 and used the 11.1 CUDA `ptxas.exe` file instead and it still didn't work. Instead I got `SubProcess ended with return code: 0`, and it still stalled after 2 batches.

**Comment by Zhaopudark** [comment]
Same with me

**Comment by not-Ian** [comment]
@amahendrakar any other ideas?

**Comment by iperov** [comment]
https://github.com/tensorflow/tensorflow/issues/45648

**Comment by not-Ian** [comment]
> #45648
> 
> this fix solves the problem
> ![102058534-11408180-3e09-11eb-9bea-839dce2e9870 1](https://user-images.githubusercontent.com/8076202/102778886-c7c5d880-43ac-11eb-87a9-35284b68883f.jpg)
> 
> but that is not a solution of course. I hate tensorflow.

WOW I think this fixed it. Weird fix, but thanks a lot.

Do you know - did changing this remove 'CUDA' from the Task Manager?

Thanks again

**Comment by iperov** [comment]
don't close the issue. 
This shit should be fixed in tensorflow.

**Comment by not-Ian** [comment]
yeah, you're right

**Comment by psycho2012** [comment]
It does not work for me.

**Comment by psycho2012** [comment]
Solution for Keras, remove activation='relu' from CONv2D

`conv1 = Conv2D(denseList[0], 3, activation='relu', padding='same')(inputs)`

to 

`
conv1 = Conv2D(denseList[0], 3, padding='same')(inputs)`
`conv1 = Activation("relu")(conv1) `

**Comment by iperov** [comment]
@psycho2012 lmao, it is not a solution. Solution is to migrate to pytorch for example.

**Comment by sushreebarsa** [comment]
Was able to replicate the issue in TF 2.6.0-dev20210602,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/301523f986df999a53a3e6aa799839ed/untitled185.ipynb#scrollTo=Bd7j7vhJXlkV)..Thanks !

**Comment by jvishnuvardhan** [comment]
@not-Ian I am not able to reproduce the issue with `TF2.5`. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/5926d407d86e4aa1559f26fa3e430561/untitled185.ipynb) and let me know if I am missing anything. Please note that i ran your code in colab (not on Windows). 

Can you please check with `TF2.5` and let us know whether the issue is persisting or not. Thanks!
